backend/ownerSizeTotalWorker.js

```diff
@@ -0,0 +1,270 @@
+        const newOwnerID = await sbp('chelonia.db/get', `_private_owner_${ownerID}`, { bypassCache: true })
+        // Found the ultimate owner
+        if (!newOwnerID) break
+        if (!depth) {
+          throw new Error('Exceeded max depth looking up owner for ' + contractID)
+        }
+        ownerID = newOwnerID
+      }
+      // Aggregate delta for the ultimate owner.
+      const [val, set] = ultimateOwners.get(ownerID) || [0, new Set()]
+      set.add(contractID)
+      ultimateOwners.set(ownerID, [val + delta, set])
+    }))
+
+    // Phase 2: Apply the aggregated deltas to the database and clean up
+    // the temporary index.
+    await Promise.all(Array.from(ultimateOwners).map(async ([id, [totalDelta, contributingResources]]) => {
+      // Apply the calculated delta to the owner's total size record in the DB
+      await updateSize(id, `_private_ownerTotalSize_${id}`, totalDelta)
+      // Remove the processed resource IDs from the persistent temporary index
+      await removeFromTempIndex(Array.from(contributingResources))
+    }))
+
+    // Reschedule the task for the next interval
+    setTimeout(sbp, TASK_TIME_INTERVAL, 'backend/server/computeSizeTaskDeltas')
+  },
+  /**
+   * Selector: 'backend/server/computeSizeTask'
+   * Task to perform a full recalculation of total owner sizes.
+   * Triggered on startup if `updatedSizeList` is populated (from
+   * persistent index).
+   * Processes resource IDs from `updatedSizeList`.
+   */
+  'backend/server/computeSizeTask': async function () {
+    // Capture the current list and clear it
+    const resourcesToRecalculate = Array.from(updatedSizeList)
+    // Clear should happen *after* successful processing per owner below:
+    // // updatedSizeList.clear();
+
+    // Map to store: ultimateOwnerID -> Set<resourceIDs belonging to this owner>
+    const ultimateOwners = new Map()
+
+    // Phase 1: Find the ultimate owner for each resource ID needing recalc
+    await Promise.all(resourcesToRecalculate.map(async (contractID) => {
+      let ownerID = contractID
+      // --- Potentially Slow Owner Lookup Loop ---
+      for (let depth = 128; depth >= 0; depth--) {
+        // Fetch the immediate owner.
+        const newOwnerID = await sbp('chelonia.db/get', `_private_owner_${ownerID}`, { bypassCache: true })
+        // Found the ultimate owner
+        if (!newOwnerID) break
+        if (!depth) {
+          throw new Error('Exceeded max depth looking up owner for ' + contractID)
+        }
+        ownerID = newOwnerID
+      }
+      // Group resources by their ultimate owner.
+      const resources = ultimateOwners.get(ownerID)
+      if (resources) {
+        resources.add(contractID)
+      } else {
+        ultimateOwners.set(ownerID, new Set([contractID]))
+      }
+    }))
+
+    // Phase 2: For each owner, fetch all their resources and calculate the
+    // total size.
+    await Promise.all(Array.from(ultimateOwners).map(async ([resourceID, contractIDs]) => {
```

Comment:
Could you please rename `resourceID` to `ownerID` to be consistent here with the previous code?